### PR TITLE
Make getContext return Context instead of hash

### PR DESCRIPTION
**Pull request by tomaszdurka**

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@vogdb please review

File: lib/job/manager.js [hunk 1]
```diff
@@ -58,19 +59,21 @@ JobManager.prototype.stop = function() {
 JobManager.prototype._processJobFile = function(filePath) {
   var self = this;
   var job;
```
Comment (vogdb) [inline lib/job/manager.js]: `var job` is redundant now here

File: lib/job/model/abstract.js [hunk 2]
```diff
@@ -133,7 +138,7 @@ AbstractJob.prototype._runJobScript = function(commandText) {
       return result.stdout.toString();
     })
     .catch(function(error) {
-      serviceLocator.get('logger').warn('Failed job process', _.extend({command: commandText}, self.getContext()));
+      serviceLocator.get('logger').warn('Failed job process', context);
```
Comment (vogdb) [inline lib/job/model/abstract.js]: should we add here `error` for `context`?

File: lib/job/model/audioroom-recording.js [hunk 3]
```diff
@@ -35,11 +36,11 @@ AudioroomRecordingJob.prototype._run = function() {
     .then(function(mp3File) {
       return self._audioConvert(wavFile, mp3File)
         .then(function() {
-          return serviceLocator.get('cm-application').importMediaStreamArchive(channelUid, mp3File)
+          return serviceLocator.get('cm-application').importMediaStreamArchive(channelUid, mp3File, context)
         })
         .then(function() {
           return unlink(wavFile).catch(function(error) {
-            serviceLocator.get('logger').error('Removing wave file failed', _.extend({error: error}, self.getContext()));
+            serviceLocator.get('logger').error('Removing wave file failed', context.clone().extend({error: error}));
```
Comment (vogdb) [inline lib/job/model/audioroom-recording.js]: why `context.clone`?

Comment (tomaszdurka) [inline lib/job/model/audioroom-recording.js]: We don't want to modify the outer context, but create one within error context/closure.

File: lib/job/model/abstract.js (hunk 2 above)
Comment (tomaszdurka) [inline lib/job/model/abstract.js]: Yep, why nots.

File: lib/job/manager.js (hunk 1 above)
Comment (tomaszdurka) [inline lib/job/manager.js]: Ok.

File: lib/job/model/audioroom-recording.js (hunk 3 above)
Comment (vogdb) [inline lib/job/model/audioroom-recording.js]: but `self.getContext()` returns a new context each time

**Comment by vogdb** [comment]
My main question is have you tested it locally?

**Comment by vogdb** [comment]
Nonetheless changes are good to merge

File: lib/job/model/audioroom-recording.js (hunk 3 above)
Comment (tomaszdurka) [inline lib/job/model/audioroom-recording.js]: Yes we could reuse, just thought it's good practice to not.

**Comment by tomaszdurka** [comment]
@vogdb Please review

**Comment by vogdb** [comment]
- Why JanusHttpClient.prototype.detach and JanusHttpClient.prototype.stopStream are different in handling of context? `requestContext`
- `this.fields = _.extend(this.fields, context.fields);` => `_.extend(this.fields, context.fields);`
- `this.extend(context.fields); return this;` => `return this.extend(context.fields);`

**Comment by njam** [comment]
- Add docu to README how to pass context variables

**Comment by njam** [comment]
- Change parameter to: `?context={"userId":<userId>,…}` (value will be URL encoded)

**Comment by njam** [comment]
- Rename "fooId" to "foo" in the context?

**Comment by tomaszdurka** [comment]
Will move last comment to another issue.

**Comment by tomaszdurka** [comment]
@vogdb 

- Kinda streamlined
- Ok
- Ok

**Comment by vogdb** [comment]
my notes are resolved.

**Comment by tomaszdurka** [comment]
@vogdb please review last commits

**Comment by vogdb** [comment]
lgtm

**Pull request merged**